ToolChain/Assembler/Assembler.py

```diff
@@ -56,79 +50,90 @@ def __init__(self, inputFile=None, outputFile=None):
-
-        validInternalRef = []   # This will be used to validate for good external reference
-                                # This works because all reference (internal and external) end up
-                                # listed in the internal references anyway.
-
-        # Deal with internal symbols
-        fileContent += b"<InternalSymbols>"
-        for reference in self.parser.referenceDict:
-            if EXPORTED_REFERENCE_INDICATOR not in reference:
-                validInternalRef.append(reference)
-                fileContent += b"<refName>" + reference.encode("utf-8") + b"</refName>"
-                fileContent += b"<refAdd>" + struct.pack(">I", self.parser.referenceDict[reference]) + b"</refAdd>"
-
-                # Update global offsets would it be required
-                # This is required since all global offsets are set to 0 at the moment they are parsed
-                wouldBeExported = EXPORTED_REFERENCE_INDICATOR + " " + reference
-                if wouldBeExported in self.parser.referenceDict.keys():
-                    self.parser.referenceDict[wouldBeExported] = self.parser.referenceDict[reference]
-
-        fileContent += b"</InternalSymbols>"
-
-        # External symbols allow for a file to be linked with another file
-        fileContent += b"<ExternalSymbols>"
-        for reference in self.parser.referenceDict:
-            if EXPORTED_REFERENCE_INDICATOR in reference:
-                isValid = False
-                shortReference = reference.split()[1]
-                for internalReference in validInternalRef:
-                    if shortReference == internalReference:
-                        isValid = True
-                        break
-
-                if isValid:
-                    refName = reference.split()[1]
-                    fileContent += b"<refName>" + refName.encode("utf-8") + b"</refName>"
-                    fileContent += b"<refAdd>" + struct.pack(">I", self.parser.referenceDict[reference]) + b"</refAdd>"
+
+        parser = Parser()                    # Parser object which will return the binary code for instructions
+        masterString = b""                   # String that will eventually be our output file
+        xmlstring = b""                      # String that will contain the "xml" data at the beginning of the .o file
+        globalList = []                      # Temporary list to hold the names of global(external) symbols
+        externalList = []                    # Master list of all external symbols to be used by linker
+        internalList = []                    # Master list of all internal symbols
+        offset = 0                           # Offset used to calculate a label's offset from the beginning of the file
+        lineno = 0                           # Keeps track of which line we're parsing to pinpoint errors
+
+        operand = 0                          # Flag for an operand, meaning parser found a valid instruction
```

Comment:
operand is usually the term used to define the elements that are part of an instruction. For example:
MOV $A $B
$A is the first operand and $B is the second operand.
Can we change this to "instruction" or similar? Also, these flags could be prefixed with the "flag" prefix to make it clear  that they are related even if someone does not read the comment.